File: app/src/main/java/com/lazysoul/kotlinwithandroid/ui/MainMvpPresenter.java
package com.lazysoul.kotlinwithandroid.ui;

import com.lazysoul.kotlinwithandroid.common.BaseMvpPresenter;
import com.lazysoul.kotlinwithandroid.common.BaseMvpView;
import com.lazysoul.kotlinwithandroid.datas.Todo;

import java.util.List;

/**
 * Created by Lazysoul on 2017. 7. 9..
 */

interface MainMvpPresenter<MvpView extends BaseMvpView> extends BaseMvpPresenter<MvpView> {

    List<Todo> loadTotoList();

    List<Todo> search(String keyword);

    void update(Todo todo);

    void remove(Todo todo);

    void delete(Todo todo);
}

File: app/src/main/kotlin/com/lazysoul/kotlinwithandroid/ui/main/MainMvpPresenter.kt
package com.lazysoul.kotlinwithandroid.ui.main

import com.lazysoul.kotlinwithandroid.common.BaseMvpPresenter
import com.lazysoul.kotlinwithandroid.common.BaseMvpView

/**
 * Created by Lazysoul on 2017. 7. 9..
 */

interface MainMvpPresenter<in MvpView : BaseMvpView> : BaseMvpPresenter<MvpView> {

    fun createTodoSamples()

    fun loadTodoList()

    fun insert(id: Int, body: String)

    fun update(id: Int, body: String)

    fun searchQuery(text: String)

    fun searchFinish()

    fun checked(id: Int, isChecked: Boolean)
}
